### PR TITLE
feat: CS dashboard tabs

### PR DESCRIPTION
**Pull request by pauldambra**
motion towards https://github.com/PostHog/posthog/issues/16028

(not the greatest demo since it uses the same dashboard for all the options so the IDs stack instead of being replaced)

tries to do the minimum so @camerondeleone and @simfish85 can test this

* store a bag linking scene to dashboard on Team
* let people choose a dashboard for Person and Group team
* when loading that dashboard save the id of the current person or group as a filter on the dashboard (see notes below)

## notes

 * should remove items from the bag if dashboards are deleted (I sort of hate Django 1-to-M keys and its simple to check on delete signal)
 * needs https://github.com/PostHog/posthog/pull/16653 before we can filter by person ID, so it filters by person email
 * if we want to roll this out further we could have a template for person and group dashboards so that you start out with our suggested dashboard, otherwise a better empty state to let people know what to set up
 * saving/replacing the filters is a bit kludgy, we should let dashboards have temporary filters without saving them
 * _really_ this needs dashboard filters to affect HogQL insights too

### CODE REVIEW

**Comment by pauldambra** [comment]
@benjackwhite did a first pass last night. Switched to a many-to-one relationship adding a new Model to User because ofc the user can have per team preferences too and a simple bag was looking messy

File: posthog/api/user.py [hunk 1]
```diff
@@ -333,6 +381,17 @@ def request_email_verification(self, request, **kwargs):
 
         return Response({"success": True})
 
+    @action(methods=["POST"], detail=True)
```
Comment (benjackwhite) [inline posthog/api/user.py:384]: Being pedantic but REST-wise isn't this more of a PUT operation?

File: posthog/models/user_scene_dashboard_choice.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from django.db import models
+
+from .utils import UUIDModel
+
+
+class UserSceneDashboardChoice(UUIDModel):
```
Comment (benjackwhite) [inline posthog/models/user_scene_dashboard_choice.py:6]: Potentially bike shedding...

Once we have Notebooks I could imagine the same functionality. Do we want to pre-emptively name the model something more generic? `UserSceneRelationChoice` or something?

Comment (benjackwhite) [reply posthog/models/user_scene_dashboard_choice.py:6]: Thinking about SharingConfiguration being a similar thing

File: frontend/src/scenes/persons/PersonDashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    const { dashboardLogicProps } = useValues(personDashboardLogic)
+
+    // TODO: needs https://github.com/PostHog/posthog/pull/16653 so we can filter by person ID
+    const { dashboard } = useValues(dashboardLogic(dashboardLogicProps))
+    const { setProperties } = useActions(dashboardLogic(dashboardLogicProps))
+    useEffect(() => {
+        if (dashboard && person) {
+            // `dashboard?.filters.properties` is typed as `any` but it's a list...
+            const current = Array.isArray(dashboard?.filters.properties) ? dashboard?.filters.properties : []
+            const hogQLPersonFilter = `person.properties.email = '${person.properties.email}'`
```
Comment (benjackwhite) [inline frontend/src/scenes/persons/PersonDashboard.tsx:24]: Is this how it works? Only via email property? 
I set this up locally and tried for user without that set and the whole things was pretty broken. I think if email is required we should make that super obvious. _or_ allow them to specify the user property they care about.

If we keep this flagged, happy for that to be follow up but it was a jarring experience when I tried it.

File: frontend/src/scenes/persons/PersonDashboard.tsx
Comment (benjackwhite) [inline frontend/src/scenes/persons/PersonDashboard.tsx:1]: UI Placement of button is weird... Not sure I have a strong suggestion of where it should go though
Screenshot 2023-08-02 at 09 12 27

File: frontend/src/scenes/persons/PersonDashboard.tsx (hunk 3 above)
Comment (pauldambra) [reply frontend/src/scenes/persons/PersonDashboard.tsx:24]: Yep, this is a "for now while behind a flag" thing. I think Simon and Cam use email already.

I'll move the TODO down a few lines `// TODO: needs https://github.com/PostHog/posthog/pull/16653 so we can filter by person ID`

File: frontend/src/scenes/persons/PersonDashboard.tsx
Comment (pauldambra) [reply frontend/src/scenes/persons/PersonDashboard.tsx:1]: Yeah, in a perfect world I guess it'd be in the tab bar. Plonked it there to match the home page position for muscle memory

File: posthog/api/user.py (hunk 1 above)
Comment (pauldambra) [reply posthog/api/user.py:384]: Are we creating a choice or updating a choice, it felt like over-kill to have both routes

Think of it as creating the command to apply the choice and POST makes sense 🙃

**Review by benjackwhite — APPROVED** [review]
Nothing scary. Lets go

**Pull request merged**